From c76cd14c4b928c713e29e8db8ad0889f08810f5d Mon Sep 17 00:00:00 2001
From: Mike Zeller <mike@mikezeller.net>
Date: Wed, 23 Oct 2019 16:19:15 +0000
Subject: [PATCH] OS-8008 bhyve on AMD should report SVM as disabled Reviewed
 by: Patrick Mooney <pmooney@pfmooney.com> Reviewed by: Jason King
 <jason.king@joyent.com> Approved by: Jason King <jason.king@joyent.com>

---
 usr/src/cmd/bhyve/xmsr.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/usr/src/cmd/bhyve/xmsr.c b/usr/src/cmd/bhyve/xmsr.c
index 7fe4804a2e..994445b3e3 100644
--- a/usr/src/cmd/bhyve/xmsr.c
+++ b/usr/src/cmd/bhyve/xmsr.c
@@ -205,6 +205,17 @@ emulate_rdmsr(struct vmctx *ctx, int vcpu, uint32_t num, uint64_t *val)
 			*val = 1;
 			break;
 
+#ifndef	__FreeBSD__
+		case MSR_VM_CR:
+			/*
+			 * We currently don't support nested virt.
+			 * Windows seems to ignore the cpuid bits and reads this
+			 * MSR anyways.
+			 */
+			*val = VM_CR_SVMDIS;
+			break;
+#endif
+
 		default:
 			error = -1;
 			break;
-- 
2.17.2 (Apple Git-113)

